elfi/examples/lorenz.py

```diff
@@ -0,0 +1,331 @@
+    initial_state : ndarray
+    n_obs : int, optional
+        Number of observed variables
+    f : float, optional
+        Force term
+    phi : float, optional
+        This value is used to express stochastic forcing term. It should be configured according
+        to force term and eventually impacts to the result of eta.
+        More details in Wilks (2005) et al.
+    total_duration : float, optional
+
+    Returns
+    -------
+    m : elfi.ElfiModel
+
+    """
+    simulator = partial(forecast_lorenz, initial_state=initial_state, f=f, n_obs=n_obs, phi=phi,
+                        total_duration=total_duration)
+
+    if not true_params:
+        true_params = [2.0, 0.1]
+
+    m = elfi.ElfiModel()
+
+    y_obs = simulator(*true_params, random_state=np.random.RandomState(seed_obs))
+    sumstats = []
+
+    elfi.Prior('uniform', 0.5, 4, model=m, name='theta1')
+    elfi.Prior('uniform', 0, 0.3, model=m, name='theta2')
+    elfi.Simulator(simulator, m['theta1'], m['theta2'], observed=y_obs, name='Lorenz')
+
+    sumstats.append(elfi.Summary(mean, m['Lorenz'], name='Mean'))
+
+    sumstats.append(elfi.Summary(var, m['Lorenz'], name='Var'))
+
+    sumstats.append(elfi.Summary(autocov, m['Lorenz'], name='Autocov'))
+
+    sumstats.append(elfi.Summary(cov, m['Lorenz'], name='Cov'))
+
+    sumstats.append(elfi.Summary(xcov, m['Lorenz'], False, name='CrosscovLeft'))
+
+    sumstats.append(elfi.Summary(xcov, m['Lorenz'], True, name='CrosscovRight'))
+
+    elfi.Discrepancy(chi_squared, *sumstats, name='d')
+
+    return m
+
+
+def mean(x):
+    """Return the mean of Y_{k}.
+
+    Parameters
+    ----------
+    x : np.array of size (b, n, m) which is (batch_size, time, n_obs)
+
+    Returns
+    -------
+    np.array of size (b,)
+        The computed mean of one vector in statistics.
+
+    """
+    return np.mean(x[:, 0, :], axis=1)
+
+
+def var(x):
+    """Return the variance of Y_{k}.
+
+    Parameters
+    ----------
+    x : np.array of size (b, n, m) which is (batch_size, time, n_obs)
+
+    Returns
+    -------
+    np.array of size (b,)
+        The computed variance of one vector in statistics.
+
+    """
+    return np.var(x[:, 0, :], axis=1)
+
+
+def cov(x):
+    """Return the covariance of Y_{k} with its neighbour Y_{k+1}.
+
+    Parameters
+    ----------
+    x : np.array of size (b, n, m) which is (batch_size, time, n_obs)
+
+    Returns
+    -------
+    np.array of size (b,)
+        The computed covariance of one vector in statistics.
+
+    """
+    x_next = np.roll(x[:, 0, :], -1, axis=1)
+    return np.mean((x[:, 0, :] - np.mean(x[:, 0, :], keepdims=True, axis=1)) *
+                   (x_next - np.mean(x_next, keepdims=True, axis=1)),
+                   axis=1)
+
+
+def xcov(x, side=False):
+    """Return the cross-covariance of Y_{k} with its neighbours from previous time steps.
+
+    Parameters
+    ----------
+    x : np.array of size (b, n, m) which is (batch_size, time, n_obs)
+
```

Comment:
`side` undocumented. Perhaps it could be renamed to e.g. `prev` (`next` is reserved).